### PR TITLE
Salto 2563 - Modify change validators to supports maps

### PR DESCRIPTION
**Pull request by Shulik95**
_modify change validators to support maps instead of lists_

---

_None_

---
_Release Notes_: 
_Netsuite Adapter_:
* remove_list_item and remove_customlist_item now work with maps instead of lists

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts [hunk 1]
```diff
@@ -17,23 +17,25 @@ import {
   ChangeValidator, getChangeData, isModificationChange, InstanceElement, isInstanceChange,
   ModificationChange,
   ElemID,
+  Values,
 } from '@salto-io/adapter-api'
-import { collections } from '@salto-io/lowerdash'
+// import { collections } from '@salto-io/lowerdash'
```
Comment (omrilit) [inline packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts:22]: leftover :)

File: packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts [hunk 2]
```diff
@@ -17,23 +17,25 @@ import {
...
 import { CUSTOM_LIST, NETSUITE } from '../constants'
 
-const { makeArray } = collections.array
+// const { makeArray } = collections.array
```
Comment (omrilit) [inline packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts:25]: leftover :)

File: packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts [hunk 3]
```diff
@@ -17,23 +17,25 @@ import {
...
 
 const hasItemRemoval = (change: ModificationChange<InstanceElement>): boolean => {
-  const beforeCustomList = change.data.before
-  const afterCustomList = change.data.after
+  const beforeCustomList: Record<string, Values> = change.data.before.value.customvalues
+    ?.customvalue
+  const afterCustomList: Record<string, Values> = change.data.after.value.customvalues?.customvalue
   const afterItemsScriptIds = new Set(
-    makeArray(afterCustomList.value.customvalues?.customvalue).map(item => item.scriptid)
+    Object.entries(afterCustomList).map(item => item[1].scriptid)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts:35]:
```suggestion
    Object.keys(afterCustomList)
```

File: packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts [hunk 4]
```diff
@@ -17,23 +17,25 @@ import {
...
   )
-  const beforeItems = makeArray(beforeCustomList.value.customvalues?.customvalue)
-  return beforeItems.some(beforeItem => !afterItemsScriptIds.has(beforeItem.scriptid))
+  const beforeItems = Object.entries(beforeCustomList)
+  return beforeItems.some(beforeItem => !afterItemsScriptIds.has(beforeItem[1].scriptid))
```
Comment (omrilit) [inline packages/netsuite-adapter/src/change_validators/remove_customlist_item.ts:38]:
```suggestion
  return Object.keys(beforeCustomList).some(beforeItemScriptId => !afterItemsScriptIds.has(beforeItemScriptId))
```

File: packages/netsuite-adapter/src/change_validators/remove_list_item.ts [hunk 5]
```diff
@@ -48,8 +48,9 @@ const getScriptIdsUnderLists = async (instance: InstanceElement):
       if (path.isAttrID()) {
         return WALK_NEXT_STEP.SKIP
       }
-      if (Array.isArray(value)) {
-        wu(value)
+      if (path.createParentID().name === 'customvalues') {
+        const valueToArray = Object.entries(value).map(elem => elem[1])
+        wu(valueToArray)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/change_validators/remove_list_item.ts:53]: This filter is not only about `customvalues` but about lists (maps) in general.
Maybe we should also rename the filter?
We should probably use the Salto type of the field (`Map`) and check if it has scriptId field.

**Review by omrilit — COMMENTED** [review]
When looking at it, it seems like there is a duplication between the filters and `removeCustomListItemValidator` is redundant since its logic is contained in the `removeListItemValidator` changeValidator, so it can be removed.

In addition, the fact that the UTs didn't fail after you changed the code is concerning and I assume that it means that we lack of coverage in this filter.

In addition

File: packages/netsuite-adapter/src/change_validator.ts [hunk 6]
```diff
@@ -15,6 +15,7 @@
 */
 import _ from 'lodash'
 import { ChangeError, ChangeValidator } from '@salto-io/adapter-api'
+// import removeCustomListItemValidator from './change_validators/remove_customlist_item'
```
Comment (omrilit) [inline packages/netsuite-adapter/src/change_validator.ts:18]: I think that all of this file can be removed and not commented out, right?

Comment (Shulik95) [reply packages/netsuite-adapter/src/change_validator.ts]: 👍

**Review by omrilit — APPROVED** [review]
Great work!

**Pull request merged**